### ISSUE TITLE
submodule virtualenv - fatal: reference is not a tree

### ISSUE THREAD

**Issue by dyndna** [PacificBiosciences/FALCON-integrate#40]
Trying to install FALCON-integrate with following steps:

```
git clone git://github.com/PacificBiosciences/FALCON-integrate.git
cd FALCON-integrate
git checkout 0.4.0  # or whatever version you want
make init
make virtualenv
make check
make -j install
make test  # to run a simple one
```

At `make init` step, I get following error:

```
git submodule update --init --depth=1 virtualenv
Submodule 'virtualenv' (git://github.com/pb-cdunn/virtualenv) registered for path 'virtualenv'
Cloning into 'virtualenv'...
remote: Counting objects: 52, done.
remote: Compressing objects: 100% (47/47), done.
remote: Total 52 (delta 0), reused 19 (delta 0), pack-reused 0
Receiving objects: 100% (52/52), 1.76 MiB | 69.00 KiB/s, done.
Checking connectivity... done.
fatal: reference is not a tree: 8e9cb4a7c34a12762a0ef8b2c5003ded4346cd49
Unable to checkout '8e9cb4a7c34a12762a0ef8b2c5003ded4346cd49' in submodule path 'virtualenv'
make: *** [init] Error 1
```

> git version 2.3.5

**Comment by pb-cdunn** [comment]
Oh, sorry. With `--depth=1` you get only the latest commit. I just updated our **virtualenv** submodule with the latest code from **pypa** (the owner). That will break `git submodule update --depth=1 virtualenv` for _all_ existing tags. I don't have a solution for that, except docs, which I will update. For now, just run that command yourself, without `--depth=1`. Let us know if you hit a snag after that.

I'll remove `--depth=1` from the `master`, but because it is a large repo, that will take a couple of minutes. Someday, we will switch to downloading a tarball for that repo, but not today.

**Comment by dyndna** [comment]
It works:

```
git clone git://github.com/PacificBiosciences/FALCON-integrate.git
cd FALCON-integrate
git checkout 0.4.2  # or whatever version you want
git submodule update --init virtualenv
make init

make virtualenv
make check
make -j install
make test  # to run a simple one
```

> if `make virtualenv` throws an error on not finding `libpython2.7.so.1.0`, it is most likely that your current python (`which python`) `lib/` directory is not in `LD_LIBRARY_PATH`. See fix at http://stackoverflow.com/a/7880519/1243763

**Comment by pb-cdunn** [comment]
Thanks for that other link too.

**Comment by pb-cdunn** [comment]
I've fixed this in `0.4.0` and later.
